assets/js/modules/search-console/components/dashboard/SearchFunnelWidgetGA4/Overview/index.js

```diff
@@ -164,6 +176,29 @@ const Overview = ( {
 		! error &&
 		! showRecoverableAnalytics;
 
+	const { setValue } = useDispatch( CORE_UI );
+	const { triggerOnDemandTour } = useDispatch( CORE_USER );
+	useEffect( () => {
+		if (
+			toursAreOnCooldown ||
+			!! shownTour ||
+			! showGA4 ||
+			dashboardType !== DASHBOARD_TYPE_MAIN
+		) {
+			return;
+		}
+
+		setValue( 'forceInView', true );
+		triggerOnDemandTour( ga4Reporting );
```

Comment:
Rather than baking the trigger directly into this component, we might simply set a flag that it's loaded. That's how we handled a similar condition with the tour for the All Traffic widget: https://github.com/google/site-kit-wp/blob/eb03c347ada0b26172f02f39ac130deeb61c5960/assets/js/feature-tours/all-traffic-widget.js#L52

I think on-demand tours are intended more to be in response to a user-action/event rather than an arbitrary condition (although it could be for that too). If this is all-around better, I'm fine with that too, but worth considering the above route as well since that's what we've done before.

Reply:
That approach looks more complicated to me, plus it locks down feature tours until the all traffic widget is loaded which may not happen at all if the user opens the dashboard page with the monetisation section selected and doesn't scroll. Not sure that it is a good idea to have multiple feature tours that can block feature tours if some conditions are not met. What do you think?

Reply:
SGTM. I think that part may have been done before we had the lazy loading where not all sections were active at once, but it's hard to remember. We can keep as-is, I just wanted to make sure we had considered the other.